conans/client/rest/rest_client_v2.py

```diff
@@ -246,13 +246,16 @@ def remove_packages(self, ref, package_ids=None):
         for ref in refs:
             assert ref.revision is not None, "remove_packages needs RREV"
             if not package_ids:
-                url = self.router.remove_all_packages(ref)
-                response = self.requester.delete(url, auth=self.auth, headers=self.custom_headers,
-                                                 verify=self.verify_ssl)
-                if response.status_code != 200:  # Error message is text
-                    # To be able to access ret.text (ret.content are bytes)
-                    response.charset = "utf-8"
-                    raise get_exception_from_error(response.status_code)(response.text)
+                # Check there are packages to remove
+                package_search_url = self.router.search_packages(ref)
```

Comment:
The ``search_packages`` is  quite slow, tbh, I am not very excited about this approach. I would say:

- This seems something to be fixed in the server side. I think it is a bug and can be fixed, it only belongs to the Conan API
- In the meantime, I would try to workaround it in the client side with a try - except 404 and only checking the ``search_packages()`` in that case. But annotating clearly it is a temporary workaround.